libraries/chain/authorization_manager.cpp

```diff
@@ -193,14 +194,14 @@ namespace eosio { namespace chain {
          p.last_updated = creation_time;
          p.auth         = std::move(auth);
 
-         if (auto dm_logger = _control.get_deep_mind_logger()) {
+         if (auto dm_logger = _control.get_deep_mind_logger(); dm_logger && !trx_is_transient) {
```

Comment:
There are two `get_deep_mind_logger` methods. One on `controller` and one on `transaction_context`. It took me a bit to realize that `transaction_context::get_deep_mind_logger` returns null for transient where `controller::get_deep_mind_logger` does not. 

I would recommend either renaming one of them or maybe change `controller::get_deep_mind_logger` to take a `bool trx_is_transient` flag so that it is obvious they do something different.

Reply:
Would actually prefer no default on that. In general I dislike default unless it causes a lot of code changes. In this case I think you need to pass in the bool on every call.

Reply:
Note how the non-default argument made it much easier to see wrong use. I doubt I would have caught these new issues below if it was defaulted.

Reply:
Thanks! That is a good practice.